src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -3,580 +3,381 @@
+        public void SetAndCheckBasePriority(ProcessPriorityClass exPriorityClass, int priority)
         {
-            // Create a process that reads one line from std in, writes it back out to the console, and exits cleanly
-            Process p = new Process();
-            p.StartInfo.FileName = s_ProcessName;
-            p.StartInfo.Arguments = "input";
-            return p;
+            s_process.PriorityClass = exPriorityClass;
+            s_process.Refresh();
+            Assert.Equal(priority, s_process.BasePriority);
         }
 
-        static Process CreateProcess()
+        [Fact]
+        public void Process_BasePriority()
         {
-            // Create a process that immediately exits
-            Process p = new Process();
-            p.StartInfo.FileName = s_ProcessName;
-            return p;
+
+            //SetAndCheckBasePriority(ProcessPriorityClass.RealTime, 24);
+            SetAndCheckBasePriority(ProcessPriorityClass.High, 13);
+            SetAndCheckBasePriority(ProcessPriorityClass.Idle, 4);
+            SetAndCheckBasePriority(ProcessPriorityClass.Normal, 8);
         }
 
-        [Fact, ActiveIssue(541)]
-        public static void Process_BasePriority()
+        public void DelayTask(double delayInMs)
         {
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
-            {
-                Assert.Equal(8, p.BasePriority);
-            }
-            finally
+            var t = Task.Run(async delegate
             {
-                p.Kill();
-                p.WaitForExit();
-            }
+                await Task.Delay(TimeSpan.FromMilliseconds(delayInMs));
+            });
+            t.Wait();
         }
 
-        private static bool s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+        public void DelayTask()
+        {
+            DelayTask(50D);
+        }
 
-        [Fact]
-        public static void Process_EnableRaiseEvents()
+        public void StartAndKillProcessWithDelay(Process p)
         {
-            // Test behavior when EnableRaisingEvent = true;
-            // Ensure event is called.
-            Process p = CreateProcessInfinite();
-            p.EnableRaisingEvents = true;
-            p.Exited += p_Exited;
             p.Start();
+            DelayTask();
             p.Kill();
             p.WaitForExit();
-            Assert.True(s_Process_EnableRaiseEvents_isExitedEventHandlerCalled, String.Format("Process_CanRaiseEvents0001: {0}", "isExited Event not called when EnableRaisingEvent is set to true."));
+        }
 
-            s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+        [Fact]
+        public void Process_EnableRaiseEvents()
+        {
+            {
+                // Test behavior when EnableRaisingEvent = true;
+                // Ensure event is called.
+                s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
```

Comment:
Why is this global state necessary?  Couldn't we just hook up a lambda to p.Exited that accesses a local?  I realize you didn't introduce this field, but as long as we're cleaning up the tests, it'd be nice to clean this up as well.  Things like this will make it easier for bugs to slip in.